decision-process/README.md

```diff
@@ -0,0 +1,77 @@
+---
+type: decision
+stakeholders:
+    - sig-architecture
+    - team-horizon
+    - team-team
+issue:
+---
+
+# RFC Decision Making Process
+
+This RFC follows on from the RFC repository README (see [here](https://github.com/giantswarm/rfc/blob/b3f4d2af6ff8965e542baae7d0e8fadb9163f356/README.md)).
+
+This RFC aims to expand on the current RFC process by adding a lightweight decision making process.
+The aim of this decision making process is to help enable, improve, and streamline technical decision making at Giant Swarm.
+
+## RFC Types
+
+To allow for a decision making process without removing the current free-form nature of RFCs, RFC types are proposed. The following changes will be introduced:
+
+- All RFCs will have a YAML header.
+- The YAML header will contain the following mandatory fields:
+    - `type`
+
+The 'default' type is `general`, and has no other requirements, to allow for free form writing. All currently merged RFCs would fall under this type, and will be moved to this type.
+
+The following is an entire and valid example of a YAML header for an RFC:
+
+```
+---
+type: general
+---
+```
+
+## Decision Making Process
+
+The two main aims of the decision making process are to ensure that the relevant groups have seen and approve the decision, and that the implementation of the decision is being tracked.
+
+To that end, the `type` field can also be `decision`. It is understood that when a `decision` RFC has been merged, that the decision has been taken and agreed on.
```

Comment:
I see a contradiction here. RFC is "request for comments", nor for making decision. When, after the discussion, a decision is made, it should be recorded in ADR (if it was about architecture), PDR (if about product and its features) or as a part of our docs (standards, guides).

Reply:
see https://github.com/giantswarm/rfc/pull/70#discussion_r1266904630 - i think RFCs could/should replace ADRs in the long-term. PDRs / docs are somehow out of scope for this, i feel

Reply:
True, the terms RFC/ADR/PDR have a strong connotation in the IT worl, and we already have those terms in use.

\<strong-opinion\>
Neither of the terms is great, and in Giant Swarm's usual way, we should make the best use out of the solution, no matter how it's called. I don't mind finding a new word, but it doesn't matter as long as we have a great and speedy decision process. For comparison, we don't use ["OKRs"](https://rework.withgoogle.com/guides/set-goals-with-okrs/steps/introduction/) and I'm happy that TeamTeam never even mentioned that horribly-working concept to us, but we track quarterly goals in simple "3 stones", and if we ever strongly believe that a little more tracking or complexity is needed, we will adapt. We didn't just take the off-the-shelf – and I have to say it again: _totally not working_ – concept and throw it at employees. From what I saw since I joined, many smart people had great ideas to take the good parts of something and apply it to the swarm, from which we have benefited a lot compared to companies that struggle to apply some allegedly-standard concept like RFCs.

One of the reasons why others have failed is to split RFCs from ADRs. Why is it so hard to put everything necessary for a decision into one easily-findable Markdown file? Why do companies write pages-long articles that could be summarized in some bullet points? Why do tech companies write down implementation details in an ADR when either 1) it could be a few bullet points or 2) the implementation will anyway happen on the spot as with regular work tasks, is therefore not plannable, and should simply go into issues that refer to the one central RFC?
\</strong-opinion\>

Suggestion: we decide for one term, which could even be a funny, GS-specific thing like RFS (Request for Swarm), and use it only in the way defined in _this RFC_, without trying to use anything from how other companies use RFCs, or how we had previously utilized the RFC/ADR/PDR model/intranet/repo. Let's start from scratch. One CI-checked YAML+Markdown format, clear state diagram (which transitions can happen, which fields are necessary in each), one file (_this one_, moved into a more discoverable place, replacing the README) documenting the process.

With that in mind, I unfortunately don't see a decision making process in this document (at time of reading i.e. 42daf33bc6c87a86064d65779fd9befb674596f0), despite the heading "Decision Making Process". Right now, we're defining a file format for the decisions, but we need to write down the whole process. Most probably it will be quite concise if we only describe the different states (e.g. `open` = in review, `approved`, `deprecated`) and what is required in the file, and required to be done, in each of those. I'm offering my help to write and implement that, but based on experience, we first need to have concensus that this file-based model will work and be used by the whole company, and that we'll move existing RFCs into the new format.